File: TestApp/app/src/main/java/com/romiope/testapp/BaseActivity.java
package com.romiope.testapp;

import android.support.v7.app.AppCompatActivity;
import rx.subscriptions.CompositeSubscription;

public abstract class BaseActivity extends AppCompatActivity {

    protected CompositeSubscription subscriptions;

    @Override
    protected void onStart() {
        super.onStart();
        subscriptions = new CompositeSubscription();
    }

    @Override
    protected void onStop() {
        subscriptions.unsubscribe();
        super.onStop();
    }
}

File: app/src/main/java/com/teso/net/ui/base/BaseActivity.kt
package com.teso.net.ui.base

import android.os.SystemClock
import android.support.v7.app.AppCompatActivity
import com.teso.net.utils.Constants
import timber.log.Timber


abstract class BaseActivity : AppCompatActivity() {

    private var lastClickTime = SystemClock.elapsedRealtime()

    override fun onStart() {
        super.onStart()
        Timber.d(" ==> Activity started - ${this.localClassName}")
    }

    override fun onStop() {
        super.onStop()
        Timber.d(" <== Activity stopped - ${this.localClassName}")
    }

    fun isClickAllowed(): Boolean {
        val now = SystemClock.elapsedRealtime()
        return if (now - lastClickTime > Constants.CLICK_TIME_INTERVAL) {
            lastClickTime = now
            true
        } else
            false
    }
}